acceptancetests/update-jenkins-slave.bash

```diff
@@ -1,15 +1,20 @@
 #!/bin/bash
 set -eux
 
+#make sure we have proper known hosts
+ssh-keyscan -H bazaar.launchpad.net git.launchpad.net upload.launchpad.net ppa.launchpad.net github.com  104.130.14.184 feature-slave-a.vapour.ws feature-slave-b.vapour.ws feature-slave-c.vapour.ws feature-slave-d.vapour.ws feature-slave-e.vapour.ws long-running-test.vapour.ws trusty-slave.vapour.ws xenial-slave.vapour.ws yakkety-slave.vapour.ws zesty-slave-a.vapour.ws release-slave.vapour.ws ci-master cwr-slave.vapour.ws juju-core-slave.vapour.ws juju-core-slave-b.vapour.ws charm-bundle-slave.vapour.ws cloud-health-slave.vapour.ws pencil-slave.vapour.ws certification-slave.vapour.ws build-slave canonistack-slave.vapour.ws feline-kiss finfolk munna silcoon arm64-slave.vapour.ws s390x-slave ppc64el-slave  >>~/.ssh/known_hosts
+sort -u ~/.ssh/known_hosts > /tmp/known_hosts
+cat /tmp/known_hosts > ~/.ssh/known_hosts
 
 update_branch() {
     # Branch or pull a branch.
     local_branch=$1
     local_dir="$(basename $local_branch | cut -d ':' -f2)"
     local_path="$HOME/$local_dir"
     if [[ -d $local_path ]]; then
-        # Clear any changes that haven't been commited to the branch.
+        # Clear any changes that have not been commited to the branch.
         (cd $local_path; bzr revert; bzr pull)
+        cd ..
```

Comment:
Yea, a bit funny. I think this was needed because git operation needed to go into the local directory to perform the operation. It would be nicer to have it cd back to a specific directory, but it is required I believe.